packages/grid/_modules_/grid/hooks/root/useRows.ts

```diff
@@ -119,14 +119,18 @@ export const useRows = (
 
       // we removes duplicate updates. A server can batch updates, and send several updates for the same row in one fn call.
       const uniqUpdates = updates.reduce((uniq, update) => {
+        if (update.id == null) {
+          throw new Error('Material-UI: Missing row Id in row data update.');
```

Comment:
Maybe something in this order?

```suggestion
          throw new Error([
            'Material-UI: The data grid component requires all rows to have a unique id.',
            'A row was provided without when calling updateRowData():',
            JSON.stringify(rowData),
          ]).join('\n');
```
```